dockerfiles/Dockerfile

```diff
@@ -12,7 +12,36 @@ RUN python3 -m pip install --no-cache-dir --upgrade pip setuptools wheel && \
     python3 -m pip wheel . --wheel-dir /work/wheels
 
 
-FROM quay.io/app-sre/qontract-reconcile-base:0.7.2
+FROM quay.io/app-sre/qontract-reconcile-base:0.7.2 as dev-image
+
+ARG CONTAINER_UID=1000
+RUN useradd --uid ${CONTAINER_UID} reconcile
+
+RUN python3 -m pip install --no-cache-dir --upgrade pip setuptools wheel
+
+# Cache mount. We don't need te wheel files in the final image.
+# This COPY will create a layer with all the wheel files to install the app.
+# This layer is not needed in the final image, so we can leverage a cache mount
+# to get rid of it. Implement the cache mount un the RUN command when we are able to
+# to use the buildkit features.
+COPY --from=build-image /work/ /tmp/work/
+#RUN --mount=type=cache,target=/tmp/work/,from=build-image,source=/work \
+
+COPY . /work
+WORKDIR /work
+
+RUN python3 -m pip install --no-cache-dir --no-index --find-links=/tmp/work/wheels qontract-reconcile
+RUN python3 -m pip install -e .
+
+RUN chown -R reconcile /work && \
+    chown -R reconcile /.terraform.d
+
+USER reconcile
```

Comment:
I **love** this user. But please note there is at least one test that run only under jenkins, and the way they confirm if it's a Jenkins build is whether UID is 0. Please update `test_make.py`.

Reply:
By idea, Jenkins should not be bothered here. The idea is that `dev-image` is only used for local development, i.e., it will not be used by CI. Not sure if Im missing sth here?
CI make targets use the `--target prod-image`, which does not use this user (yet).